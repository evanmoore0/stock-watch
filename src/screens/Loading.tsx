import React, { FC, useEffect, useState } from "react";
import { TouchableOpacity, View, Text} from "react-native";
import Graphic from "../components/Graphic";
import {VictoryLine, VictoryChart} from 'victory-native'
import normalize from "../utils/normalize";
import { useLinkProps, useNavigation } from "@react-navigation/native";

import {auth} from '../../firebase-config'
import { onAuthStateChanged } from "firebase/auth";


const Loading: FC = () => {

    const navigation = useNavigation()


    // const isAuth = (user: any) => {
       
    //     // console.log(auth.currentUser?.uid)
       
    //     console.log("USERS SL:DFJS")
    //     // console.log(getAuth())

    //         if(user) {
    //             navigation.navigate("TabStack")
    //         } else {
    //             navigation.navigate("Welcome")
    //         }
    //     }
    
    useEffect(() => {

        // const subscribe = onAuthStateChanged(auth, isAuth)
        // return subscribe

    }, [])



    return(
        <View style={{flex:1}}>
            <VictoryChart>
                <VictoryLine/>
            </VictoryChart>
        </View>
    )
}

export default Loading
