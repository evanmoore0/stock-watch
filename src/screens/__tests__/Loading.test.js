import React from 'react'
import Loading from '../Loading'
import {render} from '@testing-library/react-native'

const mockedDispatch = jest.fn();

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
            dispatch: mockedDispatch
        })
    }
})


describe("Loading", () => {
    beforeEach(() => {
      // Alternatively, set "clearMocks" in your Jest config to "true"
      mockedDispatch.mockClear();
    });
  
    it("toggles navigation drawer on press", () => {
      const butt = render(<Loading />).toJSON();
      expect(butt).toMatchSnapshot()
    });
  });