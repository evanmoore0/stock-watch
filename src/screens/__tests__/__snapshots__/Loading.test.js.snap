// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading toggles navigation drawer on press 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "top": 416,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 179,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 296,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 115,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 250,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 174,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 291,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 174,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 167,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 174,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 83,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 88,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 21,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 203,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 125,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 232,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 187,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 93,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 229,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 144,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 125,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 232,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 21,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 203,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -62,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 88,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 115,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": 63,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#82C8FB",
            "borderRadius": 56,
            "height": 179,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -62,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 179,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -187,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 203,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -312,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 174,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -374,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 144,
            "width": 21,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "top": -457,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6AB664",
            "borderRadius": 56,
            "height": 174,
            "width": 21,
          }
        }
      />
    </View>
  </View>
</View>
`;
